src/app/tests/suites/include/ConstraintsChecker.h

```diff
@@ -245,6 +245,20 @@ class ConstraintsChecker
         return CheckConstraintMaxValue(itemName, current.Value(), static_cast<T>(expected));
     }
 
+    template <typename T, typename U, std::enable_if_t<!std::is_enum<T>::value, int> = 0>
+    bool CheckConstraintNotValue(const char * itemName, const chip::app::DataModel::Nullable<T> & current,
+                                 const chip::app::DataModel::Nullable<U> & expected)
+    {
+        if (current == expected)
+        {
+            std::string expectedString(expected.HasValue() ? expected.Value() : "NULL");
+            std::string currentString(current.HasValue() ? current.Value() : "NULL");
```

Comment:
I'm a little surprised this compiles if T is `int16_t`, say: we have a `?:` expression where one arm is an int and the other is a `const char *`....  What types was this tested with?

Could we just add a TestCluster test that exercises this as part of this PR?

Reply:
I only tested with what GRL reported as failing. I added additional tests to the TestCluster.